### PR TITLE
Track average backup running time and pass to intercom

### PR DESCRIPTION
**Pull request by pdewouters**
Now that we are measuring how long backups run (see #428) for we should start tracking the average time and passing it to Intercom.

### CODE REVIEW

**Comment by pdewouters** [comment]
screenshot 2015-01-05 15 18 52

**Comment by pdewouters** [comment]
- [ ] add some unit tests

File: classes/class-schedule.php [hunk 1]
```diff
@@ -965,6 +967,79 @@ protected function do_action( $action ) {
...
+	}
+
+	/**
+	 * Calculates the average run time for this schedule.
+	 *
+	 * @return string
+	 */
+	public function get_schedule_average_duration() {
+
+		if ( ! isset( $this->options['backup_run_count'] ) || ! isset( $this->options['backup_run_count'] )  ) {
```
Comment (willmot) [inline classes/class-schedule.php]: `! isset( $this->options['backup_run_count']` is duplicated in the if statement.

File: classes/class-schedule.php [hunk 2]
```diff
@@ -965,6 +967,79 @@ protected function do_action( $action ) {
...
+			return '0';
+		}
+
+		if ( 0 === (int) $this->options['backup_run_count'] ) {
+			return '0';
```
Comment (willmot) [inline classes/class-schedule.php]: Would be good to show something other than `0` if the backup hasn't run yet as `0` makes it seem like the average run time is `0` which is probably misleading. Instead I'd show something like `N/A`.

**Pull request merged**